packages/rspeedy/plugin-react/src/loaders.ts

```diff
@@ -81,34 +114,87 @@ export function applyLoaders(
         rule.uses.delete(CHAIN_ID.USE.SWC)
         const swcLoaderRule = uses[CHAIN_ID.USE.SWC]!
           .entries() as Rspack.RuleSetRule
-        const swcLoaderOptions = swcLoaderRule
-          .options as Rspack.SwcLoaderOptions
+
+        const swcLoaderOptions = getMainThreadSwcLoaderOptions(
+          swcLoaderRule.options as Rspack.SwcLoaderOptions, 
+          mainThreadTransformOptions, 
+          mainThreadCompatOptions,
+        )
+
         rule.use(CHAIN_ID.USE.SWC)
           .merge(swcLoaderRule)
-          .options(
-            {
-              ...swcLoaderOptions,
-              jsc: {
-                ...swcLoaderOptions.jsc,
-                target: 'es2019',
-              },
-            } satisfies Rspack.SwcLoaderOptions,
-          )
+          .options(swcLoaderOptions)
       })
-      .use(LAYERS.MAIN_THREAD)
-        .loader(ReactWebpackPlugin.loaders.MAIN_THREAD)
-        .options({
-          compat,
-          enableRemoveCSSScope,
-          inlineSourcesContent,
-          isDynamicComponent: experimental_isLazyBundle,
-          shake,
-          defineDCE,
-        })
-      .end()
 
     // Clear the Rsbuild default loader.
     // Otherwise, the JSX will be transformed by the `builtin:swc-loader`.
     rule.uses.clear()
   })
 }
+
+function getMainThreadSwcLoaderOptions(
+  options: Rspack.SwcLoaderOptions,
+  mainThreadTransformOptions: SwcPluginReactLynxOptions,
+  mainThreadCompatOptions: CompatVisitorConfig | false,
+): Rspack.SwcLoaderOptions {
+  const swcLoaderOptions = {
+    ...options,
+    jsc: {
+      ...options.jsc,
+      target: 'es2019',
+      experimental: {
+        ...options.jsc?.experimental,
+        plugins: [
+          ...(options.jsc?.experimental?.plugins ?? []),
+          [
+            require.resolve('@lynx-js/swc-plugin-react-lynx'),
```

Comment:
We should not resolve `@lynx-js/swc-plugin-react-lynx` since it's a devDependency. Shall we use `@lynx-js/react/transform/swc-plugin-react-lynx` instead?